### PR TITLE
Handle QA feedback provided

### PR DESCRIPTION
**Pull request by KostiantynFandeliuk**
**Bug fixes identified by QA**

- Internal task number - [USF-1281](https://jira.corp.adobe.com/browse/USF-1281)
- External task number - EDS-232

1. Reset password URL has user email in it. Not big deal but I have not seen this in any other websites while password reset. (Dont remember, but i provided this feedback some where, please ignore if this is already discussed).
2. Combined Auth Modal overlay does not close on hitting Esc key.
3. Even after login, Combined Auth header button still there.
4. Cart, Checkout, PDP does load well in Mobile device but Combined Auth Modal overalys, SignIn Underlay, My account Underlay, SignIn text, My account page after login does not look good in mobile device.

https://github.com/hlxsites/boilerplate-commerce-dropins/assets/85105163/ab58b74f-f0b5-420e-af2b-14e4cf3a4aea

https://github.com/hlxsites/boilerplate-commerce-dropins/assets/85105163/fdd53cef-d206-45fa-8c2c-c19c031f8b6a

https://github.com/hlxsites/boilerplate-commerce-dropins/assets/85105163/dfe2b5b4-9d41-40d5-a4cc-4edfca891621

Test URLs:
https://qa-auth-dropin--boilerplate-commerce-dropins--hlxsites.hlx.live/
https://qa-auth-dropin--boilerplate-commerce-dropins--hlxsites.hlx.live/customer/login
https://qa-auth-dropin--boilerplate-commerce-dropins--hlxsites.hlx.live/customer/forgotpassword

### CODE REVIEW

**Comment by KostiantynFandeliuk** [comment]
@sirugh Could you do a code review for me, please? Thank you.

File: blocks/header/renderAuthDropdown.js [hunk 1]
```diff
@@ -37,19 +39,20 @@ function renderSignIn(element) {
...
+        primaryButtonText: 'My Account',
+        secondaryButtonText: 'Logout',
+        onPrimaryButtonClick: () => {
+          window.location.href = '/customer/account';
+        },
+        onSecondaryButtonClick: async () => {
+          await authApi.revokeCustomerToken();
+          window.location.href = '/';
+        },
+      }),
```
Comment (fnhipster) [inline blocks/header/renderAuthDropdown.js:55]: Are these meant to be "slots"? If so, we need to use the Slot API instead for consistency. For example, [here is how PDP implement slots like "Title" ](https://git.corp.adobe.com/magento-datalake/storefront-pdp/blob/d4857537f19d658d813d078c12727b6a1db9a573/src/containers/ProductDetails/ProductDetails.tsx#L309)

Storefront (boilerplate) Implementation would look something like

```js
slots: {
  successNotificationForm: (ctx) => {
  // root elem
   const elem = document.createElement('div');

   // renders container
    const container = authRenderer.render(SuccessNotification, {
        ...
        messageText: ctx.dictionary.Auth.SuccessNotification.message,
        ...
    })(elem);

    // adds container into slot
    ctx.appendChild(elem);
  }
```

By using Slots, the developer can leverage data from the `context` including locale dictionary and lifecycle.

**Review by fnhipster — CHANGES_REQUESTED** [review]

File: blocks/header/renderAuthDropdown.js (hunk 1 above)
Comment (fnhipster) [reply blocks/header/renderAuthDropdown.js:55]: After reviewing further, these might not be actual `slots` and the problem might be a misalignment on the demo implementation in the boilerplate. Let's ignore my comment above and perhaps revisit on the Boilerplate instead.

**Review by fnhipster — APPROVED** [review]

File: blocks/header/renderAuthDropdown.js (hunk 1 above)
Comment (KostiantynFandeliuk) [reply blocks/header/renderAuthDropdown.js:55]: Thank you for your response. I will finish the implementation requested by @sirugh  and make an additional push to this PR. (refactor Dropdown)

**Comment by KostiantynFandeliuk** [comment]
**Example after refactor**

https://github.com/hlxsites/boilerplate-commerce-dropins/assets/85105163/10abe531-ca89-405c-baf8-f49688827c48

https://github.com/hlxsites/boilerplate-commerce-dropins/assets/85105163/bfc932d9-7c97-4416-9d81-06f946cb977a

@sirugh could you check please ? Thank you.

**Comment by sirugh** [comment]
If i click sign in first then the search, sometimes the search shows up and sign in goes away, and sometimes search shows up behind sign in. When search does show up behind sign in, I cant close the sign in modal. To repro, you can just click and open and close sign in and search modal a few times from the top right buttons.

File: blocks/header/header.js [hunk 2]
```diff
@@ -73,20 +71,13 @@ function toggleAllNavSections(sections, expanded = false) {
  * @param {*} forceExpanded Optional param to force nav expand behavior when not null
  */
 function toggleMenu(nav, navSections, forceExpanded = null) {
-  const expanded = forceExpanded !== null
-    ? !forceExpanded
-    : nav.getAttribute('aria-expanded') === 'true';
+  const expanded =
```
Comment (sirugh) [inline blocks/header/header.js:74]: Please fix this.

Comment (KostiantynFandeliuk) [reply blocks/header/header.js]: Fixed

**Review by sirugh — APPROVED** [review]

**Pull request merged**